src/main/java/com/parsing/parsers/parserlot/ParserLotService.java

```diff
@@ -0,0 +1,118 @@
+                if (parsePDFLink(document).equals(NOT_FOUND_HREF)) {
+                    lot.setStatus(Status.CREATED);
+                } else {
+                    lot.setStatus(Status.PARSED);
+                }
+                lotRepository.save(lot);
+                log.info("parsed URL = " + url);
+            }
+        }
+    }
+
+    private String parseDK(Document document) {
+        String source = document.getElementsByClass("tender-date padding-left-more").toString();
+        Pattern pattern = Pattern.compile("\\d{8}");
+        Matcher matcher = pattern.matcher(source);
+        String dk = null;
+        if (matcher.find()) {
+            dk = matcher.group();
+        }
+        return dk;
+    }
+
+    private String parsePDFLink(Document document) {
+        Element element = document.select("table[class=table table-striped margin-bottom prev]").last();
+        if (element == null){
+            element = document.select("table[class=table table-striped margin-bottom prev]").first();
+        }
+        if (element != null) {
+            List<Element> elements = element.getElementsByAttribute("href");
+            if (elements.size() > 1) {
+                Element source = elements.get(elements.size() - 1);
+                return source.attr("href");
+            } else if (elements.size() == 1) {
+                Element source = elements.get(0);
+                return source.attr("href");
+            } else {
+                return NOT_FOUND_HREF;
+            }
+        }
+        return NOT_FOUND_HREF;
+    }
+
+    LocalDate parseDate(String uri) {
+        String date = uri.replaceAll("https://prozorro.gov.ua/tender/UA-", "").substring(0, 10);
+        return LocalDate.parse(date);
+    }
+
+    BigDecimal parsePrice(Document document) {
+        Element element = document.getElementsByClass("green tender--description--cost--number").first();
+        String source = element.getElementsByTag("strong").toString();
+        String price = source.replace("<strong>", "")
+                .replace("<span class=\"small\">UAH</span></strong>", "")
+                .replace(" ", "")
+                .replace(",", ".");
+        return new BigDecimal(price);
+    }
+
+    public List<String> prepareURI() {
+        final String MAIN_URI = "https://prozorro.gov.ua/tender/UA";
+        final String DASH = "-";
+        final String END_URI = "-a";
+        Constants constants = new Constants();
+        List<String> listDates = constants.createDates();
+        List<String> listLots = constants.createCounter();
+        List<String> listUrls = new ArrayList<>();
+        List<String> temp = listDates.stream().map(x -> MAIN_URI.concat(DASH).concat(x).concat(DASH)).toList();
+        for (String datesCounter : temp) {
+            for (String intCounter : listLots) {
+                listUrls.add(datesCounter.concat(intCounter).concat(END_URI));
+            }
+        }
+        log.debug("prepare URI complete");
```

Comment:
you should somehow distinct this log, use some name or url or Id to make it useful, otherwise you won't understand much from this log

Reply:
ok.